libs/libc/syslog/lib_setlogmask.c

```diff
@@ -91,17 +96,29 @@ uint8_t g_syslog_mask = LOG_ALL;
 int setlogmask(int mask)
 {
   uint8_t oldmask;
+
+#if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)
```

Comment:
The better fix is either:
1.remove the critical section or
2.replace with mutex/semphare
The mix just make the logic more complex and unmaintainable

Reply:
> 
> 
> The better fix is either:
> 1.remove the critical section or
> 2.replace with mutex/semphare
> The mix just make the logic more complex and unmaintainable

A pthread mutex cannot be across task groups (or at least should not because it breaks the usaage model).  "pshared" semaphores (only) would be okay but would probably need to created via sem_open().

Test and set could probably also be used but is not available in all architectures.

Reply:
Thanks and sorry for my incomplete understanding for NuttX task group model. I found that pthread_mutex is not prefer for this case.

As the Greg's last comment in ML, Do I just need to remove the critical section in this case because setlogmask() have no need to concern about MT-safe ?